src/main/scala/service/GameService.scala

```diff
@@ -20,12 +18,9 @@ trait GameService {
   def getLastGameResult(): Either[Error, FinalResult]
 }
 
-class GameServiceImpl extends GameService {
-
-  val gameRepository: GameRepository = new GameRepositoryImpl()
-  val moveSerialization = CaseEnumSerialization.apply[Move]
-
-  private def provideCPUMove(): Move = Random.shuffle(moveSerialization.values).head
+class GameServiceImpl(
+  val gameRepository: GameRepository = new GameRepositoryImpl(), 
+  val cpuMoveStrategy: CpuMoveStrategy = new CpuMoveStrategyImpl()) extends GameService {
```

Comment:
@veej I understand your scares, usually in this case I have adopted a system for dependency injection framework, but I'm not documented enough about best practices for scala DI before writing this - my fault 😥 -  I'll try to refactor avoid defaults.
I agree with you about **val** never access required it's useless

Reply:
Usually, we construct the needed dependencies in the Main file and pass them manually between the services.
This will also help you if the same dependency is used by two or more services (you can create it only once and pass the same instance to every service that needs it)